_posts/2021-11-12-github-copilot.md

```diff
@@ -0,0 +1,128 @@
+
+My interest in this tool comes from always wanting to improve efficiency and learning, so I applied for technical preview, and got accepted, and have now used this for a few weeks for various personal projects.
+
+## What can Github Copilot do for you?
+
+Well, quite a lot. Once you have installed the plugin and the Copilot has been enabled for you, you can start writing code, and it will be automatically filled in with code suggestions. You can ignore them and keep on coding as usual, or press the TAB key to accept the suggestions. You can also press Ctrl+Enter to get up to 10 different suggestions for the problem.
+
+Copilot is using the code you have written as a hint for the code suggestions. If you started writing a function, you'll get nice suggestions for the function, based on the name and parameters. If you started defining a variable, you get suggestions on how to populate it. You can also write comments, to explain in English what you want to accomplish.
+
+Let's see how Copilot could help you implement a simple temperature system conversion function. Easy mode is to start defining a function, let's do this in Python. Once you start writing the function name, you get the suggestion for implementation (in gray). You can accept it by pressing the TAB key.
+
+![Github Copilot generating Celsius to Fahrenheit function](/img/github-copilot/copilot_celsius_to_fahrenheit_cropped.gif)
+*Copilot is creating Celsius to Fahrenheit function based on function name*
+
+What if you did not like that suggestion, but would like to see more ideas? Well, you can press Ctrl+Enter to see more suggestions. Let's do that.
+
+![More suggestions for Celsius to Fahrenheit function](/img/github-copilot/celsius_to_fahrenheit_suggestions.png)
+*Copilot is giving me 9 more options for the implementation*
+
+How about those pesky unit tests? Can we generate a good base for a unit test? Let's write a comment and see if we get a suggestion.
+
+![Github Copilot generating a Test for Celsius to Fahrenheit function](/img/github-copilot/copilot_generate_test_from_comment_cropped.gif)
+*Copilot is creating a test for my function*
+
+Of course, the final question is: Can it handle coding challenges typically used in the work interviews? I grabbed a random task from a google search and put it in the comments. Then I pressed Ctrl+Enter for suggestions.
+
+![Nailing the work inteview challenges?](/img/github-copilot/copilot_interview_questions.png)
+*Copilot is implementing answer to work interview challenge*
+
+Oops! I suppose we best ask to turn off the Copilot for the interview challenges. :)
+
+## Where Copilot will fail you
+
+The current version of Copilot is mainly getting its hints from the code you write. If you provide nicely structured code, short well-named functions, it works quite intelligently. If you provide huge blobs of code, badly named things, obscure or impossible comments, you will get quite weird suggestions. Aka, garbage in, garbage out.
```

Comment:
Aka, garbage in, garbage out -> "In other words, feed Copilot garbage, you get garbage back."